python/hsfs/core/vector_server.py

```diff
@@ -143,43 +201,89 @@ def setup_online_store_sql_client(
+                features=entity.features,
+            )
+        )
+        # This logic needs to move to the above engine init
+        reset_online_rest_client = False
+        online_store_rest_client_config = None
+        if isinstance(options, dict):
+            reset_online_rest_client = options.get(
+                self.RESET_ONLINE_REST_CLIENT_OPTIONS_KEY, reset_online_rest_client
+            )
+            online_store_rest_client_config = options.get(
+                self.ONLINE_REST_CLIENT_CONFIG_OPTIONS_KEY,
+                online_store_rest_client_config,
+            )
+
+        online_store_rest_client.init_or_reset_online_store_rest_client(
+            optional_config=online_store_rest_client_config,
+            reset_client=reset_online_rest_client,
+        )
+
     def get_feature_vector(
         self,
         entry: Dict[str, Any],
         return_type: Optional[str] = None,
         passed_features: Optional[Dict[str, Any]] = None,
         vector_db_features: Optional[Dict[str, Any]] = None,
-        td_embedding_feature_names: Optional[set[str]] = None,
         allow_missing: bool = False,
+        force_rest_client: bool = False,
+        force_sql_client: bool = False,
     ) -> Union[pd.DataFrame, pl.DataFrame, np.ndarray, List[Any], Dict[str, Any]]:
         """Assembles serving vector from online feature store."""
-        if passed_features is None:
-            passed_features = {}
-        if td_embedding_feature_names is None:
-            td_embedding_feature_names = set()
-        # get result row
-        _logger.debug("get_feature_vector Online SQL client")
-        serving_vector = self.online_store_sql_client.get_single_feature_vector(entry)
-        if vector_db_features:
-            serving_vector.update(vector_db_features)
-        # Deserialize complex features
-        _logger.debug("Deserializing complex features")
-        serving_vector = self.deserialize_complex_features(
-            serving_vector, td_embedding_feature_names
+        online_client_choice = self.which_client_and_ensure_initialised(
+            force_rest_client=force_rest_client, force_sql_client=force_sql_client
         )
+        rondb_entry = self.validate_entry(
+            entry=entry,
+            allow_missing=allow_missing,
+            passed_features=passed_features,
+            vector_db_features=vector_db_features,
+        )
+        if len(rondb_entry) == 0:
+            _logger.debug("Empty entry for rondb, skipping fetching.")
+            serving_vector = {}  # updated below with vector_db_features and passed_features
+        elif online_client_choice == self.DEFAULT_ONLINE_STORE_REST_CLIENT:
+            _logger.debug("get_feature_vector Online REST client")
+            serving_vector = self.online_store_rest_client_engine.get_single_feature_vector(
+                rondb_entry,
+                allow_missing=allow_missing,
+                drop_missing=True,
```

Comment:
What is the funcitonality of drop_missing as an explicit parameter online_store_rest_client_engine